tool_watershed/watershed_analysis_networkx.py

```diff
@@ -21,7 +22,9 @@
 class Graph3Di:
     def __init__(
         self,
+        model,
```

Comment:
Please use type annotation

Reply:
I think this can be left out altogether; see my later comment suggesting a simpler implementation